spec/lib/tasks/change_history_spec.rb

```diff
@@ -0,0 +1,241 @@
+      ClimateControl.modify NOTE: "Test 99" do
+        Rake::Task["change_history:edit"].invoke(edition.content_id, id_to_edit)
+      end
+
+      change_history[1][:note] = "Test 99"
+
+      expect(edition.reload.revision.metadata_revision.change_history)
+        .to eq(change_history.map(&:with_indifferent_access))
+    end
+
+    it "calls the edition updater" do
+      ClimateControl.modify NOTE: "Test 99" do
+        Rake::Task["change_history:edit"].invoke(edition.content_id, id_to_edit)
+      end
+
+      change_history[1][:note] = "Test 99"
+
+      expect(Tasks::EditionUpdater).to have_received(:call)
+    end
+
+    it "raises an error when no note is supplied" do
+      expect { Rake::Task["change_history:edit"].invoke(edition.content_id, "abc123") }
+        .to raise_error("Expected a note")
+
+      ClimateControl.modify NOTE: "" do
+        expect { Rake::Task["change_history:edit"].invoke(edition.content_id, "abc123") }
+          .to raise_error("Expected a note")
+      end
+    end
+
+    it "raises an error when the change history ID does not exist" do
+      ClimateControl.modify NOTE: "Test 99" do
+        expect { Rake::Task["change_history:edit"].invoke(edition.content_id, "abc123") }
+          .to raise_error("No change history entry with id abc123")
+      end
+    end
+  end
+
+  describe "change_history:add" do
+    let(:change_history) do
+      [
+        {
+          id: SecureRandom.uuid,
+          public_timestamp: "2020-02-20T16:00:00+00:00",
+          note: "Test 1",
+        },
+      ]
+    end
+
+    let(:edition) do
+      create(:edition,
+             locale: "en",
+             change_history: change_history)
+    end
+
+    before { Rake::Task["change_history:add"].reenable }
+
+    it "adds a change history entry note" do
+      ClimateControl.modify NOTE: "Test 99", TIMESTAMP: "2020-10-22 09:30" do
+        Rake::Task["change_history:add"].invoke(edition.content_id)
+      end
+
+      expect(edition.reload.revision.metadata_revision.change_history.first["note"])
+        .to eq("Test 99")
+      expect(edition.reload.revision.metadata_revision.change_history.last["note"])
+        .to eq("Test 1")
+    end
+
+    it "calls the edition updater" do
+      ClimateControl.modify NOTE: "Test 99", TIMESTAMP: "2020-10-22 09:30" do
+        Rake::Task["change_history:add"].invoke(edition.content_id)
+      end
+
+      expect(Tasks::EditionUpdater).to have_received(:call)
+    end
+
+    it "adds a change history entry note in the correct order" do
+      ClimateControl.modify NOTE: "Test 99", TIMESTAMP: "2019-01-01 00:00" do
+        Rake::Task["change_history:add"].invoke(edition.content_id)
+      end
+
+      expect(edition.reload.revision.metadata_revision.change_history.first["public_timestamp"])
+        .to eq("2020-02-20T16:00:00+00:00")
+      expect(edition.reload.revision.metadata_revision.change_history.last["public_timestamp"])
+        .to eq("2019-01-01T00:00:00+00:00")
```

Comment:
This could be a bit more concise:

```suggestion
      change_notes = edition.reload.change_history.map { |c| c["note"] }
      expect(change_notes).to eq(["Test 1", "Test 99"])
```

You might want to consider renaming your change notes to give them more meaning. For example one could be "February update" another could be "January update", which would be easier to parse than the dates and working with phrasing like "Test 99" (where you might question does 99 mean something?)

Reply:
Fixed